### DISCUSSION TITLE
[ORM] Recommended way to update an empty column?

### DISCUSSION THREAD

**Discussion by Diego-Hernandez-Moodys** [sqlalchemy/sqlalchemy#7684] [Usage Questions] [answered]
I have a query, column A with varchar(200), and column B with null data (empty) I'd like to populate with processed data from column A.

```python3
query = (
   session.query(Table.column_pk, Table.column_a, Table.column_b)
   .filter(blah blah)
   )
```

And I call a function that process all items in column A
```python3
gen_column_b = ( proces(entry.column_b) for entry in query )
```

How would I batch insert update these row values? I.e., how do I set this entire set of data to Column B in this query, then commit it?

I've tried doing them one by one, but get the Null RowProxy error
```python3
# entry.column_b is None
for entry in query:
   entry.column_b = process(entry.column_a)
session.commit()

>>>
AttributeError: 'Row' object has no attribute 'column_b'
```

I've tried filtering the original query by the primary key

```python3
gen_data = ( [entry.column_pk, process(entry.column_b)] for entry in query )
for data in gen_data:
    query.filter(Table.column_pk == data[0]).update({"column_b" : data[1] })
session.commit()

>>>
sqlalchemy.exc.ArgumentError: subject table for an INSERT, UPDATE or DELETE expected, got Column('column_pk', ..., primary_key=True, nullable=False).
```

**Comment by zzzeek** [comment, marked as answer]
an UPDATE is against the whole table, not the columns, so format your query like this:

    session.query(Table).filter(Table.column_pk == data[0]).update({"column_b" : data[1] })

background at https://docs.sqlalchemy.org/en/14/orm/session_basics.html#update-and-delete-with-arbitrary-where-clause

**Reply by Diego-Hernandez-Moodys** [reply]
Thank you!
